proposals/0012-introduce-reactNativeMetadata.md

```diff
@@ -0,0 +1,329 @@
+
+```json
+{
+  "reactNativeManifest": {
+    "android": {
+      "capabilities": {
+        "hermes": {
+          "enabled": true
+        }
+      }
+    },
+    "ios": {
+      "capabilities": {
+        "hermes": {
+          "enabled": true
+        }
+      }
+    }
+  }
+}
+```
+
+### React Native Release feature flagging
+
+We currently have a numer of different files where capabilities of React Native can be toggled to enable/disable features in the runtime.
+For example we have:
+* [android/gradle.properties](https://github.com/facebook/react-native/blob/main/packages/react-native/template/android/gradle.properties) to specify key-value build time properties for Android
+* [android/app/build.gradle](https://github.com/facebook/react-native/blob/main/packages/react-native/template/android/app/build.gradle) to specify build time configuration for Android
+* [ReactNativeFeatureFlags.js](https://github.com/facebook/react-native/blob/main/packages/react-native/Libraries/ReactNative/ReactNativeFeatureFlags.js) which contains runtime feature toggles for the Javascript layer.
+* [ReactFeatureFlags.java](https://github.com/facebook/react-native/blob/main/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/config/ReactFeatureFlags.java) which contains runtime feature toggles for Android.
+* [ReactNativeConfig.cpp](https://github.com/facebook/react-native/blob/main/packages/react-native/ReactCommon/react/config/ReactNativeConfig.cpp) which contains runtime feature toggles for the C++ layer.
+
+Most of those config flags are undocumented and they're not verified for invalid configuration. Specifically, some feature flags require to be toggled on multiple layers, and it's easy to forget to toggle one of them. Invalid combinations of those flags could lead to unexpected runtime behavior.
+
+We propose to define `reactNativeManifest` as the **preferred entry point** for all the user facing entry points. All the tooling should be adapted to consume the information from the `reactNativeManifest` section and have sensible defaults defined by the tooling itself.
+
+It's outside the scope of this RFC to declare all the new sections we intend to add, and we defer to the [Future proof and extensibility](#future-proof-and-extensibility) section for the process on how to include a new key to the `reactNativeManifest` section.
+
+## Excluded use cases
+
+This section lists the use cases where we believe the `reactNativeManifest` section should **not** be used.
+
+### React Native version support
+
+For the time being, we're not planning to add a `version` section to the `reactNativeManifest` section for libraries. The rationale is that we believe that the `react-native` version should be specified in either the `peerDependencies` or the `engines` section of the `package.json` of the project. This follows the idiomatic way to specify the version of a dependency in a NodeJS project.
```

Comment:
The examples below have a `version` section in the `reactNativeManifest`.  Is that a typo?